epochStart/metachain/stakingDataProvider.go

```diff
@@ -532,6 +544,48 @@ func (sdp *stakingDataProvider) GetNumOfValidatorsInCurrentEpoch() uint32 {
 	return sdp.numOfValidatorsInCurrEpoch
 }
 
+func (sdp *stakingDataProvider) updateEpochStats(validator state.ValidatorInfoHandler) {
+	validatorCurrentList := common.PeerType(validator.GetList())
+	shardID := validator.GetShardId()
+
+	if validatorCurrentList == common.EligibleList {
+		sdp.validatorStatsInEpoch.Eligible[shardID]++
+		return
+	}
+
+	if validatorCurrentList == common.WaitingList {
+		sdp.validatorStatsInEpoch.Waiting[shardID]++
+		return
+	}
+
+	validatorPreviousList := common.PeerType(validator.GetPreviousList())
+	if sdp.isValidatorLeaving(validatorCurrentList, validatorPreviousList) {
+		sdp.validatorStatsInEpoch.Leaving[shardID]++
+	}
+}
+
+func (sdp *stakingDataProvider) isValidatorLeaving(validatorCurrentList, validatorPreviousList common.PeerType) bool {
+	if validatorCurrentList != common.LeavingList {
```

Comment:
the forcedToStay validators in the current epoch for the next epoch their list is updated to Eligible or Waiting right?

Reply:
I checked an integration tests I did superficially for forced to stay nodes and checked with debug what happens in that case.

For every forced to stay epoch, those nodes will allways have:
- Current list = Leaving
- Previous List = Eligible/Waiting

So I guess it is correct, right?

Reply:
in the next epoch, do they have the CurrentList=Leaving? or the owner has to "unstake" them again because they are eligible/waiting?

Reply:
Nodes that are forced to stay will remain in system no matter what the owner does.
As I've said: 
>For every forced to stay epoch, those nodes will allways have:
> - Current list = Leaving
> - Previous List = Eligible/Waiting

Reply:
ok